src/backend/opencl/kernel/transpose.cl

```diff
@@ -15,6 +15,12 @@ T doOp(T in) {
 #define doOp(in) in
 #endif
 
+#pragma OPENCL EXTENSION all : enable
+#ifdef cl_khr_fp16
+#else
+#define half short
+#endif
```

Comment:
Interesting, why is it that we need these `#if`s only in `jit.cl` and `transpose.cl` ?

Reply:
We will probably need it in other places but those functions haven't been implemented yet.